pkg/controller/core/workload_controller.go

```diff
@@ -369,10 +373,16 @@ func (r *WorkloadReconciler) reconcileCheckBasedEviction(ctx context.Context, wl
 	log := ctrl.LoggerFrom(ctx)
 	log.V(3).Info("Workload is evicted due to admission checks")
 	if workload.HasRejectedChecks(wl) {
-		wl.Spec.Active = ptr.To(false)
-		if err := r.client.Update(ctx, wl); err != nil {
-			return false, err
+		var rejectedCheckNames []string
+		for _, check := range workload.RejectedChecks(wl) {
+			rejectedCheckNames = append(rejectedCheckNames, check.Name)
+		}
+		workload.SetDeactivationTarget(wl, kueue.WorkloadEvictedByAdmissionCheck, fmt.Sprintf("Admission check(s): %v, were rejected", rejectedCheckNames))
+		workload.SetEvictedCondition(wl, kueue.WorkloadEvictedByAdmissionCheck, "Admission check(s) were rejected")
```

Comment:
Is there a need to evict the workload here given that we have logic responsible for that?
https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/core/workload_controller.go#L191-L199

Reply:
It seems to me like this could fix integration tests